bats_ai/core/management/commands/importRecordings.py

```diff
@@ -0,0 +1,202 @@
+                            '  No activation start time in metadata, using file modification time'
+                        )
+                    )
+
+                # Create Point from latitude/longitude if available
+                point = None
+                if metadata.get('nabat_latitude') and metadata.get('nabat_longitude'):
+                    point = Point(metadata['nabat_longitude'], metadata['nabat_latitude'])
+
+                # Get grid cell ID
+                grts_cell_id = None
+                if metadata.get('nabat_grid_cell_grts_id'):
+                    try:
+                        grts_cell_id = int(metadata['nabat_grid_cell_grts_id'])
+                    except (ValueError, TypeError):
+                        pass
+
+                # Convert species list to string if present
+                species_list_str = None
+                if metadata.get('nabat_species_list'):
+                    species_list_str = ','.join(metadata['nabat_species_list'])
+
+                # Create recording
+                self.stdout.write('  Creating recording...')
+                with open(wav_file, 'rb') as f:
+                    recording = Recording(
+                        name=wav_file.name,
+                        owner=owner,
+                        audio_file=File(f, name=wav_file.name),
+                        recorded_date=recorded_date,
+                        recorded_time=recorded_time,
+                        equipment=None,  # Not in GUANO metadata
+                        grts_cell_id=grts_cell_id,
+                        recording_location=point,
+                        public=is_public,
+                        comments=metadata.get('nabat_comments'),
+                        detector=metadata.get('nabat_detector_type'),
+                        software=metadata.get('nabat_software_type'),
+                        site_name=metadata.get('nabat_site_name'),
+                        species_list=species_list_str,
+                        unusual_occurrences=metadata.get('nabat_unusual_occurrences'),
+                    )
+                    recording.save()
+
+                self.stdout.write(self.style.SUCCESS(f'  Created recording ID: {recording.pk}'))
+
+                # Generate spectrogram synchronously
+                self.stdout.write('  Generating spectrogram...')
+                try:
+                    result = recording_compute_spectrogram(recording.pk)
+                    self.stdout.write(
+                        self.style.SUCCESS(
+                            f'  Spectrogram generated (ID: {result.get("spectrogram_id")})'
+                        )
+                    )
+                except Exception as e:
+                    self.stdout.write(
+                        self.style.ERROR(f'  Failed to generate spectrogram: {str(e)}')
+                    )
+                    logger.exception('Error generating spectrogram', exc_info=e)
```

Comment:
Does this need to re-raise so the outer try/except can properly increment `failed`?

Reply:
I'll update to add it in.